src/components/dashboard/MentorDashboardPage.tsx

```diff
@@ -1,24 +1,76 @@
-import React from "react";
+import React, { useEffect, useState } from "react";
 import { useNavigate } from "react-router-dom";
 import Header from "../headers/Header";
 import { useDispatch, useSelector } from "react-redux";
 import { RootState } from "../../store";
 import { deleteUser } from "../../store/features/userSlice";
+import { Mentor } from "../../models/Model";
+import MentorService from "../../apiServices/MentorService";
+import MentorTable from "../mentor/MentorTable";
 
 const MentorDashboardPage: React.FC = () => {
   const dispatch = useDispatch();
   const navigate = useNavigate();
   const user = useSelector((state: RootState) => state.user);
+  const [isLoading, setIsLoading] = useState(false);
+  const [mentorData, setMentorData] = useState<Mentor[]>([]);
+
+  const loadMentorsData = () => {
+    setIsLoading(true);
+    MentorService.getMentor()
+      .then((data) => {
+        setMentorData(data);
+      })
+      .catch((error) => {
+        alert(
+          "Something went wrong while fetching mentors. Please refresh the page"
+        );
+        console.error(error);
+      });
+    setIsLoading(false);
+  };
+
+  useEffect(() => {
+    loadMentorsData();
+  }, []);
 
   const handleLogout = () => {
     dispatch(deleteUser());
     navigate("/login");
   };
 
+  // update mentor using existing mentor data
+  const handleEditMentor = (data: Mentor) => {
+    console.log(data);
```

Comment:
What is this function doing?

Reply:
this function i have created for edit mentor will complete it.